shop/src/components/templates/OrderTemplate.jsx

```diff
@@ -0,0 +1,183 @@
+import { comma } from '../../utils/convert'
+import Button from "../atoms/Button"
+import {useMutation} from "react-query"
+import { orderCart } from '../../services/order'
+import { useNavigate } from 'react-router-dom'
+import { useRef, useState } from 'react'
+import { useQuery } from 'react-query'
+import { getCart } from '../../services/cart'
+
+const OrderTemplate = () => {
+  const navigate = useNavigate()
+  const {data} = useQuery("carts", 
+    getCart,
+    {
+      onError: (error) => {
+        console.error("Error fetching cart data:", error);
+        navigate('/404')
+      },
+      suspense: true
+    }
+  )
+  const {products, totalPrice} = data?.data?.response
+  const [allAgree, setAllAgree] = useState(false)
+  const [agreePayment, setAgreePayment] = useState(false)
+  const [agreePollcy, setAgreePollcy] = useState(false)
+  const allAgreeRef = useRef(null)
+  const agreePaymentRef = useRef(null)
+  const agreePollcyRef = useRef(null)
+
+  const handleAllAgree = (e)=>{
+    const value = e.target.checked
+    setAgreePayment(value)
+    setAgreePollcy(value)
+
+  }
+
+  const handleAgree = (e)=>{
+    const{name, checked} = e.target
+    console.log(name, checked)
+    if(name === 'payment-agree'){
+      setAgreePayment(checked)
+    } else if(name === 'pollcy-agree'){
+      setAgreePollcy(checked)
+    }
+
+    if(!checked){
+      setAllAgree(false)
+    }
+  }
+
+  const {mutate} = useMutation({
+    mutationKey: "orders",
+    mutationFn: orderCart
+  })
+
+  const OrderItems = () =>{
+    let renderComponent = []
+    products.forEach((item)=>{
+      renderComponent.push(item.carts.map((cart)=>{
+        return <div key={cart.id} className='p-4 border bg-white'>
+          <div className="product-name font-bold">
+            <span>{`${item.productName} - ${cart.option.optionName}`}</span>
+          </div>
+          <div className="quantity">
+            <span>{comma(cart.quantity)}개</span>
+          </div>
+          <div className="price font-bold">
+            <span>{comma(cart.quantity * cart.price)}원</span>
+          </div>
+        </div>
+      }))
+    })
+    return renderComponent
+  }
```

Comment:
일단 이 로직은 별도의 컴포넌트로 분리해주시는게 좋겠습니다.
OrderTemplate이 OrderItems가 어떻게 그려지는지 구체적으로 알 필요는 없어보여요.

그리고 로직상으로도 depth를 조금 줄이고 불필요한 할당을 제외해줄 수 있겠습니다.
```suggestion
const OrderItems = () => {
  const carts = products.flatMap((product) => product.carts);
  return carts.map((cart) => (
    <div key={cart.id} className='p-4 border bg-white'>
      <div className="product-name font-bold">
        <span>{`${item.productName} - ${cart.option.optionName}`}</span>
      </div>
      <div className="quantity">
        <span>{comma(cart.quantity)}개</span>
      </div>
      <div className="price font-bold">
        <span>{comma(cart.quantity * cart.price)}원</span>
      </div>
    </div>
  ));
}
```

Reply:
그런 방법이 ..!! 
수정하겠습니다!

Reply:
수정하던 중 궁금한 점이 생겼는데, OrderItems의 파일 위치를 atoms에 두는 것이 맞을까요 ??